stingray/lombscargle.py

```diff
@@ -0,0 +1,633 @@
+        self.method = method
+        self.oversampling = oversampling
+        self._make_crossspectrum(
+            self.lc1, self.lc2, fullspec, method=method, oversampling=oversampling
+        )
+        if self.power_type == "absolute":
+            self.power = np.abs(self.power)
+            self.power_err = np.abs(self.power_err)
+            self.unnorm_power = np.abs(self.unnorm_power)
+            self.unnorm_power_err = np.abs(self.unnorm_power_err)
+        if self.power_type == "real":
+            self.power = np.real(self.power)
+            self.power_err = np.real(self.power)
+            self.unnorm_power = np.real(self.unnorm_power)
+            self.unnorm_power_err = np.real(self.unnorm_power_err)
+        self._make_auxil_pds(self.lc1, self.lc2)
+
+    def initial_checks(
+        self,
+        data1,
+        data2,
+        norm,
+        power_type,
+        dt,
+        min_freq,
+        max_freq,
+        fullspec,
+        df,
+        method,
+        oversampling,
+    ):
+        if not isinstance(norm, str):
+            raise TypeError("norm must be a string")
+
+        if not isinstance(power_type, str):
+            raise TypeError("power_type must be a string")
+
+        if norm.lower() not in ["frac", "abs", "leahy", "none"]:
+            raise ValueError("norm must be one of ['frac','abs','leahy','none']")
+
+        if power_type not in ["all", "absolute", "real"]:
+            raise ValueError("power_type must be one of ['all','absolute','real']")
+
+        if np.logical_xor(data1 is None, data2 is None):
+            raise ValueError("You can't do a cross spectrum with just one lightcurve")
+
+        if min_freq < 0:
+            raise ValueError("min_freq must be non-negative")
+
+        if max_freq is not None:
+            if max_freq < min_freq or max_freq < 0:
+                raise ValueError("max_freq must be non-negative and greater than min_freq")
+
+        if method not in ["fast", "slow"]:
+            raise ValueError("method must be one of ['fast','slow']")
+
+        if not isinstance(oversampling, int):
+            raise TypeError("oversampling must be an integer")
+
+        if not isinstance(fullspec, bool):
+            raise TypeError("fullspec must be a boolean")
+
+        if np.logical_xor(
+            not (isinstance(data1, EventList) or isinstance(data1, Lightcurve) or data1 is None),
+            not (isinstance(data2, EventList) or isinstance(data2, Lightcurve) or data2 is None),
+        ):
+            raise TypeError("One of the arguments is not of type Eventlist or Lightcurve or None")
+
+        if not (
+            isinstance(data1, EventList) or isinstance(data1, Lightcurve) or data1 is None
+        ) and (
+            not (isinstance(data2, EventList) or isinstance(data2, Lightcurve) or data2 is None),
+        ):
+            raise TypeError("Both the events are not of type Eventlist or Lightcurve or None")
+
+        if type(data1) == type(data2):
```

Comment:
Instead of raising ValueError, I would just say that the lags will not be correct. The process should actually work.